openmmtools/tests/test_storage_interface.py

```diff
@@ -42,27 +44,27 @@ def spawn_driver(path):
 def test_storage_interface_creation():
     """Test that the storage interface can create a top level file and read from it"""
     with temporary_directory() as tmp_dir:
-        test_store = tmp_dir + '/teststore.nc'
+        test_store = tmp_dir + "/teststore.nc"
         driver = spawn_driver(test_store)
         si = StorageInterface(driver)
-        si.add_metadata('name', 'data')
-        assert si.storage_driver.ncfile.getncattr('name') == 'data'
+        si.add_metadata("name", "data")
+        assert si.storage_driver.ncfile.getncattr("name") == "data"
 
 
-@tools.raises(Exception)
 def test_read_trap():
     """Test that attempting to read a non-existent file fails"""
     with temporary_directory() as tmp_dir:
-        test_store = tmp_dir + '/teststore.nc'
+        test_store = tmp_dir + "/teststore.nc"
         driver = spawn_driver(test_store)
         si = StorageInterface(driver)
-        si.var1.read()
+        with pytest.raises(KeyError):
```

Comment:
Great we capturing the actual `KeyError` exception here, instead of a generic `Exception`.